src/Workspaces/Core/Portable/Rename/IChangeNamespaceService.cs

```diff
@@ -53,5 +57,22 @@ internal interface IChangeNamespaceService : ILanguageService
         /// a no-op and original solution will be returned.
         /// </remarks>
         Task<Solution> ChangeNamespaceAsync(Document document, SyntaxNode container, string targetNamespace, CancellationToken cancellationToken);
+
+        /// <summary>
+        /// Using only the top level namespace declarations of a document, change all of them to the target namespace. Will only
+        /// use namespace containers considered valid by <see cref="CanChangeNamespaceAsync(Document, SyntaxNode, CancellationToken)"/>
+        /// </summary>
+        Task<Solution> ChangeTopLevelNamespacesAsync(Document document, string targetNamespace, CancellationToken cancellationToken);
+
+        /// <summary>
+        /// Given a set of folders from a <see cref="Document"/> build the namespace that would match
+        /// the folder structure. If a document is located in "Foo/Bar/Baz" then the namespace could be 
+        /// "Foo.Bar.Baz"
+        /// 
+        /// Returns null if the folders contain parts that are invalid identifiers for a namespace.
+        /// </summary>
+        string? TryBuildNamespaceFromFolders(IEnumerable<string> folders, ISyntaxFacts syntaxFacts);
```

Comment:
not sure why this needs to be on the interface.

Reply:
Let's discuss this in https://github.com/dotnet/roslyn/pull/41267#discussion_r412766117

Reply:
https://github.com/dotnet/roslyn/pull/41267/files#r412794993